lib/api/entities.rb

```diff
@@ -251,9 +251,18 @@ class ProjectService < Grape::Entity
       expose :push_events, :issues_events, :merge_requests_events, :tag_push_events, :note_events
       # Expose serialized properties
       expose :properties do |service, options|
-        field_names = service.fields.
-          select { |field| options[:include_passwords] || field[:type] != 'password' }.
-          map { |field| field[:name] }
+        field_names = []
+        service.fields.each do |field|
+          if field[:type] == 'fieldset'
+            field[:fields].each do |subfield|
+              if options[:include_passwords] || subfield[:type] != 'password'
```

Comment:
I would like your assistance with this ... where would I put the helper function and how would I call it here?
```
def get_field_names(field, options)
  field_names = [] 
  if field[:type] == 'fieldset'
    field[:fields].each do |sub_field|
      sub_field_names = get_field_names(sub_field, options);
      field_names << sub_field_names unless sub_field_names.empty?
      field_names.flatten!
    end
  else 
    field_names << field[:name] if options[:include_passwords] || field[:type] != 'password'
  end
  field_names
end
```